packages/frontend2/src/server/features/scaling/tvl/get-scaling-tvl-entries.ts

```diff
@@ -0,0 +1,64 @@
+import { layer2s, layer3s } from '@l2beat/config'
+import {
+  type ProjectsVerificationStatuses,
+  notUndefined,
+} from '@l2beat/shared-pure'
+import { type ImplementationChangeReport } from '../../implementation-change-report/get-implementation-change-report'
+import { getCommonScalingEntry } from '../get-common-scaling-entry'
+import { orderByTvl } from '../tvl/utils/order-by-tvl'
+import { type DetailedLatestTvl } from './utils/get-detailed-7d-tvl-breakdown'
+
+export function getScalingTvlEntries({
+  implementationChangeReport,
+  projectsVerificationStatuses,
+  tvl,
+}: {
+  implementationChangeReport: ImplementationChangeReport
+  projectsVerificationStatuses: ProjectsVerificationStatuses
+  tvl: DetailedLatestTvl
+}) {
+  const projects = [...layer2s, ...layer3s].filter(
+    (project) => !project.isUpcoming && !project.isArchived,
+  )
+
+  const entries = projects
+    .map((project) => {
+      const isVerified = !!projectsVerificationStatuses[project.id.toString()]
+      const hasImplementationChanged =
+        !!implementationChangeReport.projects[project.id.toString()]
+
+      const latestTvl = tvl.projects[project.id.toString()]
+
+      return {
+        ...getCommonScalingEntry({
+          project,
+          isVerified,
+          hasImplementationChanged,
+        }),
+        href: `/scaling/projects/${project.display.slug}/tvl-breakdown`,
+        entryType: 'scaling' as const,
+        tvl: {
+          breakdown: latestTvl?.breakdown,
+          change: latestTvl?.change,
+          totalChange: latestTvl?.totalChange,
+          associatedTokens: project.config.associatedTokens ?? [],
+          warnings: [project.display.tvlWarning].filter(notUndefined),
+        },
```

Comment:
maybe it should be this way?
```suggestion
        tvl: latestTvl ? {
          breakdown: latestTvl.breakdown,
          change: latestTvl.change,
          totalChange: latestTvl.totalChange,
          associatedTokens: project.config.associatedTokens ?? [],
          warnings: [project.display.tvlWarning].filter(notUndefined),
        } : undefined,
```